packages/components/tests/integration/components/hds/table/index-test.js

```diff
@@ -1,17 +1,224 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { render, click } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 
+// we're using this for multiple tests so we'll declare context once and use it when we need it.
+const setData = (context) => {
+  context.set('model', [
+    {
+      id: '1',
+      type: 'folk',
+      artist: 'Nick Drake',
+      album: 'Pink Moon',
+      year: '1972',
+    },
+    {
+      id: '2',
+      type: 'folk',
+      artist: 'The Beatles',
+      album: 'Abbey Road',
+      year: '1969',
+    },
+    {
+      id: '3',
+      type: 'folk',
+      artist: 'Melanie',
+      album: 'Candles in the Rain',
+      year: '1971',
+    },
+  ]);
+  context.set('columns', [
+    { key: 'artist', label: 'Artist' },
+    { key: 'album', label: 'Album' },
+    { key: 'year', label: 'Year' },
+  ]);
+  context.set('sortBy', 'artist');
+  context.set('sortOrder', 'asc');
+};
+const renderSortableTable = async () => {
+  await render(hbs`
+  <Hds::Table
+        @model={{this.model}}
+        @sortBy={{this.sortBy}}
+        @sortOrder={{this.sortOrder}}
+        @columns={{this.columns}}
+        id="data-test-table"
+      >
+        <:body as |row|>
+          <Hds::Table::Tr>
+            <td>{{row.artist}}</td>
+            <td>{{row.album}}</td>
+            <td>{{row.year}}</td>
+          </Hds::Table::Tr>
+        </:body>
+      </Hds::Table>
+  `);
+};
+
 module('Integration | Component | hds/table/index', function (hooks) {
   setupRenderingTest(hooks);
 
   test('it renders the component', async function (assert) {
     await render(hbs`<Hds::Table />`);
     assert.dom(this.element).exists();
   });
+
   test('it should render with a CSS class that matches the component name', async function (assert) {
+    await render(hbs`<Hds::Table id="data-test-table"/>`);
+    assert.dom('#data-test-table').hasClass('hds-table');
+  });
+
+  test('it should render with a CSS class appropriate for the @density value', async function (assert) {
+    await render(hbs`<Hds::Table @density="short" />`);
+    assert.dom('[data-test-table]').hasClass('hds-table--density-short');
+  });
+
+  test('it should render with a CSS class appropriate if no @density value is set', async function (assert) {
     await render(hbs`<Hds::Table />`);
-    assert.dom('[data-test-table]').hasClass('hds-table');
+    assert.dom('[data-test-table]').hasClass('hds-table--density-medium');
```

Comment:
since the `Table` has `...attributes` we can remove also this `data-test` selector and use an ID